unit/src/DependencyResolverTest.C

```diff
@@ -144,3 +144,21 @@ TEST_F(DependencyResolverTest, dependsOnTest)
   EXPECT_FALSE(_tree.dependsOn("m1", "something_else"));
   EXPECT_FALSE(_tree.dependsOn("something_else", "k0"));
 }
+
+TEST_F(DependencyResolverTest, cyclicTest)
+{
+  try
+  {
+    // Attempt to insert a dependency that results in cyclicity
+    _resolver.insertDependency("a", "b");
+    FAIL() << "missing expected exception";
+  }
+  catch (const std::exception & e)
```

Comment:
Here you want to catch the `CyclicDependencyException` object (search for it in this class). It's awesome because it's a templated class with a multimap in it so you can print out the actual values that are caught in the cycle! I just searched through MOOSE and I don't see a case were we actually use that information. I blame @aeslaughter for removing that! In either case you want to catch the right type so you don't get false positives.